tests/SharedTests/TestLeads.cs

```diff
@@ -0,0 +1,620 @@
+            var createdEntities = response.CreatedEntities;
+            Assert.IsNotNull(createdEntities);
+            Assert.IsTrue(createdEntities.Count == 0);
+        }
+
+        [TestMethod]
+        public void TestQualifyLeadRequestOpportunityCustomerIdSuccess()
+        {
+            var lead = new Lead
+            {
+                FirstName = "Test",
+                LastName = "Lead"
+            };
+            var account = new Account();
+            lead.Id = orgAdminUIService.Create(lead);
+            account.Id = orgAdminUIService.Create(account);
+
+            var request = new QualifyLeadRequest
+            {
+                Status = new OptionSetValue((int)Lead_StatusCode.Qualified),
+                LeadId = lead.ToEntityReference(),
+                OpportunityCustomerId = account.ToEntityReference()
+            };
+
+            var response = orgAdminUIService.Execute(request) as QualifyLeadResponse;
+            Assert.IsNotNull(response);
+            var createdEntities = response.CreatedEntities;
+            Assert.IsNotNull(createdEntities);
+            Assert.IsTrue(createdEntities.Count == 0);
+        }
+
+        [TestMethod]
+        public void TestQualifyLeadRequestCreateAccountSuccess()
+        {
+            var lead = new Lead
+            {
+                FirstName = "Test",
+                LastName = "Lead"
+            };
+            lead.Id = orgAdminUIService.Create(lead);
+
+            var request = new QualifyLeadRequest
+            {
+                Status = new OptionSetValue((int)Lead_StatusCode.Qualified),
+                LeadId = lead.ToEntityReference(),
+                CreateAccount = true
+            };
+
+            var response = orgAdminUIService.Execute(request) as QualifyLeadResponse;
+            Assert.IsNotNull(response);
+            var createdEntities = response.CreatedEntities;
+            Assert.IsNotNull(createdEntities);
+            Assert.IsTrue(createdEntities.Count == 1);
+        }
+
+        [TestMethod]
+        public void TestQualifyLeadRequestCreateContactSuccess()
+        {
+            var lead = new Lead
+            {
+                FirstName = "Test",
+                LastName = "Lead"
+            };
+            lead.Id = orgAdminUIService.Create(lead);
+
+            var request = new QualifyLeadRequest
+            {
+                Status = new OptionSetValue((int)Lead_StatusCode.Qualified),
+                LeadId = lead.ToEntityReference(),
+                CreateContact = true
+            };
+
+            var response = orgAdminUIService.Execute(request) as QualifyLeadResponse;
+            Assert.IsNotNull(response);
+            var createdEntities = response.CreatedEntities;
+            Assert.IsNotNull(createdEntities);
+            Assert.IsTrue(createdEntities.Count == 1);
+        }
+
+        [TestMethod]
+        public void TestQualifyLeadRequestCreateOpportunityAccountSuccess()
```

Comment:
Missing assert that lead has been qualified and that a new Opportunity have been created and points to the existing account